### PR TITLE
chore: use vite plugin for registry

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

**Review by yury-s — APPROVED** [review]

File: packages/html-reporter/src/chip.spec.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 import React from 'react';
 import { expect, test } from '@playwright/experimental-ct-react/test';
-import { AutoChip, Chip } from './chip';
+import { AutoChip, Chip as LocalChip } from './chip';
```
Comment (dgozman) [inline packages/html-reporter/src/chip.spec.tsx:19]: Why this change?

File: packages/playwright-core/bundles/utils/package.json [hunk 2]
```diff
@@ -21,13 +22,14 @@
     "progress": "2.0.3",
     "proper-lockfile": "4.1.2",
     "proxy-from-env": "1.1.0",
+    "rimraf": "^3.0.2",
```
Comment (dgozman) [inline packages/playwright-core/bundles/utils/package.json:25]: Why did this change from `3.0.2` to `^3.0.2`?

File: packages/playwright-core/bundles/utils/package.json [hunk 3]
```diff
@@ -21,13 +22,14 @@
...
     "socks-proxy-agent": "6.1.1",
     "stack-utils": "2.0.5",
-    "rimraf": "3.0.2",
     "ws": "8.4.2"
   },
   "devDependencies": {
     "@types/debug": "^4.1.7",
+    "@types/glob": "^7.2.0",
```
Comment (dgozman) [inline packages/playwright-core/bundles/utils/package.json:32]: `@types/glob` version does not match `glob` version.

File: packages/playwright-test/src/vitePlugin.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                imports.set(info.fullName, info);
+              }
+            }
+          }
+        });
+      }
+    },
+
+    transform: async (_, id) => {
+      // We ask users to put their playwright config as the main script of their html template.
```
Comment (dgozman) [inline packages/playwright-test/src/vitePlugin.ts:76]: "Playwright config" here does not mean `playwright.config.ts`, does it? Perhaps rename/update the comment?

File: packages/html-reporter/playwright.html [hunk 5]
```diff
@@ -15,15 +15,15 @@
 -->
 
 <!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (pavelfeldman) [reply packages/html-reporter/playwright.html:17]: No, it isn't, users still need to be able to configure things in their html...

File: packages/playwright-core/ThirdPartyNotices.txt [hunk 6]
```diff
@@ -7,48 +7,51 @@ This project incorporates components from the projects listed below. The origina
...
-37.	buffer-crc32@0.2.13 (https://github.com/brianloveswords/buffer-crc32)
-38.	end-of-stream@1.4.4 (https://github.com/mafintosh/end-of-stream)
-39.	extract-zip@2.0.1 (https://github.com/maxogden/extract-zip)
-40.	fd-slicer@1.1.0 (https://github.com/andrewrk/node-fd-slicer)
-41.	get-stream@5.2.0 (https://github.com/sindresorhus/get-stream)
-42.	pend@1.2.0 (https://github.com/andrewrk/node-pend)
-43.	pump@3.0.0 (https://github.com/mafintosh/pump)
-44.	yauzl@2.10.0 (https://github.com/thejoshwolfe/yauzl)
-45.	yazl@2.5.1 (https://github.com/thejoshwolfe/yazl)
+4.	brace-expansion@2.0.1 (https://github.com/juliangruber/brace-expansion)
```
[the remark this answers is not in the document]
Comment (pavelfeldman) [reply packages/playwright-core/ThirdPartyNotices.txt:10]: Fixed!

File: packages/html-reporter/src/chip.spec.tsx (hunk 1 above)
Comment (pavelfeldman) [reply packages/html-reporter/src/chip.spec.tsx:19]: To cover aliases in tests

File: packages/playwright-core/bundles/utils/package.json (hunk 3 above)
Comment (pavelfeldman) [reply packages/playwright-core/bundles/utils/package.json:32]: Best effort.

File: packages/playwright-test/src/vitePlugin.ts (hunk 4 above)
Comment (pavelfeldman) [reply packages/playwright-test/src/vitePlugin.ts:76]: Stale comment, removed.
